### PR TITLE
Feature/system notifications

### PR DESCRIPTION
**Pull request by maximilianmeier**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: the project builds without code style warnings.
- [x] I updated the documentation and models.
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~~I added (end-to-end) test cases for the new functionality.~~

### Motivation and Context
Students, Tutors, and Instructors were not notified about changes in the system, so a notification system should notify them.

### Description
This Pull Request adds the administration of system notification and notifications for exercise creations and updates. For quiz exercises, each state is sent to the client.

### Steps for Testing
1. Log in to ArTEMiS
2. Navigate to Notification Management
3. Create a new system notification, with notification date now and expire date in the future
4. Notification should be shown at the top of the container
5. create a quiz exercise
6. a notification should appear for a new exercise
7. make quiz visible and start quiz
8. two notifications should appear in the notification container

### Screenshots
image
image

### CODE REVIEW

**Comment by krusche** [comment]
Can you please rerun the new formatting settings for the changed files in this pull request? I already merged develop, so you need to pull and then make one minimal (e.g. whitespace) change in each file so that the formatter runs again. Thank you

**Comment by maximilianmeier** [comment]
> Can you please rerun the new formatting settings for the changed files in this pull request? I already merged develop, so you need to pull and then make one minimal (e.g. whitespace) change in each file so that the formatter runs again. Thank you

Done

File: src/main/webapp/app/app.main.ts [hunk 1]
```diff
@@ -6,9 +6,12 @@ ProdConfig();
 
 if (module['hot']) {
     module['hot'].accept();
+    if ('production' !== process.env.NODE_ENV) {
+        console.clear();
```
Comment (krusche) [inline src/main/webapp/app/app.main.ts]: Why do we need this code?

Comment (maximilianmeier) [reply src/main/webapp/app/app.main.ts]: this was a helper for me that clears the console when there is a hot reload. I often looked at errors that did not came from the current build but from older one. this code prevents this issue. but i can revert it again and just add it locally for me

Comment (krusche) [reply src/main/webapp/app/app.main.ts]: ok, I guess we can leave it in then

**Pull request merged**